11_notify/n0npax/pkg/puppy/rest.go

```diff
@@ -0,0 +1,131 @@
+package puppy
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+
+	middleware "github.com/anz-bank/go-course/11_notify/n0npax/pkg/middleware"
+	"github.com/gin-gonic/gin"
+)
+
+var (
+	LostPuppyURL        = ""
+	SlowRequestDuration = "2s"
+	PuppyDeleteNotifyF  = LostPuppyReq
+)
+
+// LostPuppyReq sends request to lostpuppy service
+func LostPuppyReq(id int) {
+	jsonBytes := []byte(fmt.Sprintf(`{"id":%d}`, id))
+	resp, err := http.Post(LostPuppyURL, "application/json", bytes.NewBuffer(jsonBytes)) // #nosec
```

Comment:
adding `// #nosec`

`gosec` will complain as long as `LostPuppyURL` is not const.
Due to https://stackoverflow.com/questions/24790175/when-is-the-init-function-run const is initialized before `init` and `main`, so I cannot initialize const in `init` function.

If there is a better way to solve this issue, I'd be happy to receive a hint.

Reply:
had that same issue, couldn't workout anything better either.